runtime/src/main/java/org/corfudb/runtime/DistributedCompactor.java

```diff
@@ -0,0 +1,561 @@
+
+    private boolean checkCompactionManagerStartTrigger() {
+        //This is necessary here to stop checkpointing after it has started
+        if (isCheckpointFrozen(corfuStore, this.checkpointTable) ||
+                isClient && isUpgrade(corfuStore, this.checkpointTable)) {
+            return false;
+        }
+
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            final CheckpointingStatus managerStatus = txn.getRecord(
+                    compactionManagerTable, DistributedCompactor.COMPACTION_MANAGER_KEY).getPayload();
+            txn.commit();
+            if (managerStatus == null ||
+                    managerStatus.getStatus() != StatusType.STARTED &&
+                            managerStatus.getStatus() != StatusType.STARTED_ALL) {
+                return false;
+            }
+        } catch (Exception e) {
+            log.error("Unable to acquire CompactionManager status, {}, {}", e, e.getStackTrace());
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * An entry for all the clients participating in checkpointing
+     * Tries to checkpoint tables that are marked as IDLE
+     *
+     * @return count - the number of tables checkpointed by the client
+     */
+    public int startCheckpointing() {
+        long startCp = System.currentTimeMillis();
+        log.trace("Starting Checkpointing in-memory tables..");
+        openCheckpointingMetadataTables();
+
+        if (!checkCompactionManagerStartTrigger()) {
+            log.trace("Checkpoint hasn't started");
+            return 0; // Orchestrator says checkpointing is either not needed or done.
+        }
+
+        int count = checkpointOpenedTables();
+
+        if (count <= 0) {
+            log.trace("Opened tables not checkpointed by client: {}", clientName);
+            if (count < 0) {
+                return count;
+            }
+        }
+
+        if (!isClient) {
+            log.info("{} in-memory tables checkpointed using protobuf serializer. Checkpointing remaining.", count);
+            count += checkpointUnopenedTables();
+        }
+
+        if (count > 0) {
+            long cpDuration = System.currentTimeMillis() - startCp;
+            log.info("Client {} took {} ms to checkpoint {} tables", clientName, cpDuration, count);
+
+            MicroMeterUtils.time(Duration.ofMillis(cpDuration), "checkpoint.total.client_timer",
+                    "clientName", clientName);
+            MicroMeterUtils.measure(count, "checkpoint.num_tables." + clientName);
+        }
+
+        return count;
+    }
+
+    private int checkpointUnopenedTables() {
+        int count = 0;
+
+        KeyDynamicProtobufSerializer keyDynamicProtobufSerializer = new KeyDynamicProtobufSerializer(cpRuntime);
+        cpRuntime.getSerializers().registerSerializer(keyDynamicProtobufSerializer);
+
+        List<TableName> tableNames = getAllTablesToCheckpoint();
+        for (TableName tableName : tableNames) {
```

Comment:
This can cause an NPE. Client side, this will impact subsequent executions of `DistributedClientCheckpointer::checkpointAllMyOpenedTables`.

Reply:
The DistributedClientCheckpointer will never enter the checkpointUnopenedTables() as it will not get past the isClient check in startCheckpointing() at line 207. And as far CorfuStoreCompactor, it will recover in the next cycle.
But will anyway return an emptyList for getAllTablesToCheckpoint()